lib/ansible/modules/cloud/cloudstack/cs_network_offering.py

```diff
@@ -0,0 +1,379 @@
+            'supportedservices': self.module.params.get('supported_services'),
+            'traffictype': self.module.params.get('traffic_type'),
+            'availability': self.module.params.get('availability'),
+            'conservemode': self.module.params.get('conserve_mode'),
+            'details': self.module.params.get('details'),
+            'egressdefaultpolicy': self.module.params.get('egress_default_policy'),
+            'ispersistent': self.module.params.get('persistent'),
+            'keepaliveenabled': self.module.params.get('keepalive_enabled'),
+            'maxconnections': self.module.params.get('max_connections'),
+            'networkrate': self.module.params.get('network_rate'),
+            'servicecapabilitylist': self.module.params.get('service_capability_list'),
+            'serviceofferingid': self.module.params.get('service_offering_id'),
+            'serviceproviderlist': self.module.params.get('service_provider_list'),
+            'specifyipranges': self.module.params.get('specify_ip_ranges'),
+            'specifyvlan': self.module.params.get('specify_vlan'),
+        }
+        if not self.module.check_mode:
+            res = self.cs.createNetworkOffering(**args)
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            network_offering = res['networkoffering']
+
+        return network_offering
+
+    def delete_network_offering(self):
+        network_offering = self.get_network_offering()
+
+        if not network_offering:
+            return network_offering
+
+        self.result['changed'] = True
+
+        if not self.module.check_mode:
+            res = self.cs.deleteNetworkOffering(id=network_offering['id'])
+
+        return network_offering
+
+    def update_network_offering(self, network_offering):
+        if not network_offering:
+            return network_offering
+
+        args = {
+            'id': network_offering['id'],
+            'state': self.module.params.get('state'),
+            'displaytext': self.module.params.get('display_text'),
+            'name': self.module.params.get('name'),
+            'availability': self.module.params.get('availability'),
+        }
+
+        if args['state'] == 'enabled' or args['state'] == 'disabled':
+            args['state'] = args['state'].title()
+        else:
+            del args['state']
+
+        for k, v in args.items():
+            if network_offering[k] != args[k]:
+                self.result['changed'] = True
+
+        if not self.module.check_mode:
+            res = self.cs.updateNetworkOffering(**args)
+            network_offering = res['networkoffering']
+
+        return network_offering
+
+
+def main():
+    argument_spec = cs_argument_spec()
+    argument_spec.update(dict(
+        state=dict(choices=['enabled', 'present', 'disabled', 'absent'], default='present'),
+        display_text=dict(required=True),
```

Comment:
IMHO this should only be required if `state=present`

Reply:
Right, like "service_provider_list", "supported_services" and "guest_ip_type" ?